### PR TITLE
Spec Refactor

### PR DESCRIPTION
**Pull request by darthjee**
This PR refactor the specs to use newest format

- [x] Use ```expect().to``` instead of ```.should```
- [x] Refatoração de hash_keys_changer.rb

### CODE REVIEW

File: spec/support/shared_examples/hash_keys_changer.rb [hunk 1]
```diff
@@ -1,22 +1,15 @@
...
+    { 'a' => 1, b: 2, c: { d: 3, e: 4 }, f: [{ g: 5 }, { h: 6 }] }
+  end
 
-      expect(original).to receive(:deep_dup).and_return(copy)
-      expect(copy).to receive(:change_keys!)
-      original.change_keys(recursive: true) { |k| "foo_#{k}" }
-    end
+  describe :change_keys do
 
+    it_behaves_like 'a mnethod that is able to change keys', :change_keys
```
Comment (thiagogsr) [inline spec/support/shared_examples/hash_keys_changer.rb]: mnethod *

File: spec/support/shared_examples/hash_keys_changer.rb [hunk 2]
```diff
@@ -24,44 +17,79 @@
...
-  it 'accepts block to change the keys' do
-    { 'a' => 1, b: 2 }.public_send(method) { |k| "foo_#{k}" }.should eq('foo_a' => 1, 'foo_b' => 2)
+shared_examples 'result is as expected' do
+  after do
+    expect(result).to eq(expected)
   end
 
-  it 'accepts block to change the keys' do
-    { a: 1, 'b' => 2 }.public_send(method) { |k| "foo_#{k}".to_sym }.should eq(foo_a: 1, foo_b: 2)
+  it '' do
```
Comment (thiagogsr) [inline spec/support/shared_examples/hash_keys_changer.rb]: Esse `it` vazio?

File: spec/support/shared_examples/hash_keys_changer.rb [hunk 3]
```diff
@@ -24,44 +17,79 @@
...
   end
+end
 
-  it 'applies the block recursively' do
-    { 'a' => 1, b:  { c: 3, d: 4 } }.public_send(method) { |k| "foo_#{k}" }.should eq('foo_a' => 1, 'foo_b' =>  { 'foo_c' => 3, 'foo_d' => 4 })
+shared_examples 'a mnethod that is able to change keys' do |method|
```
Comment (thiagogsr) [inline spec/support/shared_examples/hash_keys_changer.rb]: method *

**Comment by tadeusrox** [comment]
:shipit:

**Comment by thiagogsr** [comment]
:shipit:

**Pull request merged**